docs/design-documents/README.md

```diff
@@ -0,0 +1,40 @@
+# Design Process for Mbed-OS
+This document defines the software design process for Mbed-OS. Anyone developing for Mbed-OS should use this process to submit the design to get it reviewed from the broader Mbed-OS developer community.
+
+## Design documents location and organization
+* It's suggested to write one design document per change, which could be a new feature, defect fixes, tools changes etc. Do not try to address multiple changes in the same document.
+* The design document will follow the format represented in [Design Document Template](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md)
+
+* For each design submission, create a folder under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}*. The area/component would be Drivers, Platform, Connectivity, Security, Peripherals etc. 
+  If your implementation is creating new APIs, you should try to match the folder name directly to the new API, otherwise 
+  use the title of the feature as the folder name. 
+  For example, if your implementation is adding new EMAC API, you may name the folder as **New_EMAC_API** or if you are creating a new Serial Driver for Mbed-OS you may name the folder as **Serial_driver_for_Mbed-os**.
```

Comment:
I would vote for lower case naming `new_emac_api`

Reply:
Well the folder name also serve as the title, that's why I thought it should be capitalized as required.
If more people align with lower_case we can definitely change it.

Reply:
Yes - hopefully for new features within a component we can just make pull requests that update the existing design doc? Also do we really need design documents for fixes?